### ISSUE TITLE
Feature request: Support YouTube Timestamps

### ISSUE THREAD

**Issue by Mennaruuk** [bluesky-social/social-app#3943]
**Is your feature request related to a problem? Please describe.**

YouTube has a feature where a user can directly link to a particular timestamp in the video. That way, when someone watches the video, it immediately jumps to that point. It's super helpful, as it avoids having viewers skip back and forth until they find the right spot.

**Describe the solution you'd like**

I want Bluesky to include the timestamp when a timestamped video link is inserted. The current behavior is that all links with a timestamp begin from 0:00, thus ignoring the timestamp parameter.

**Describe alternatives you've considered**

I tell readers the timestamp in my posts, but this isn't ideal, and can take up some precious space when the character limit is 300.

**Example URL**

`https://youtu.be/T6eK-2OQtew?si=fs1T7_cU9xvLdhqQ&t=54`

**Additional context**

Thank you.

**Comment by mary-ext** [comment]
Are you sure the YouTube embeds aren't starting at the timestamp you've specified? because this has been fixed 4 months ago? (https://github.com/bluesky-social/social-app/pull/2565)

**Comment by mary-ext** [comment]
Can confirm they are still working, tested on Web 1.81.0 and Android 1.80.0

**Comment by Mennaruuk** [comment]
> Can confirm they are still working, tested on Web 1.81.0 (Firefox Nightly 127 and Chromium 124) and Android 1.80.0, using the link you've specified in this issue `https://youtu.be/T6eK-2OQtew?si=fs1T7_cU9xvLdhqQ&t=54`.

I see what's wrong. Links containing the `si` parameter work. Links that don't contain it don't. If you can, please try the following link:

`https://youtu.be/T6eK-2OQtew&t=51`

**Comment by mary-ext** [comment]
That link is incorrect FWIW, you'd need it to be `?t=`

**Comment by Mennaruuk** [comment]
Perhaps you're right. It's not YouTube's fault. I remove the `si` part because it's a [tracking parameter.](https://github.com/Xenorio/YTShareAntiTrack) When added, YouTube can track who is sharing videos with who. I simply resort to manually deleting the whole parameter. I know I can change the `&t=` to `?t=` every time, but I thought it would be nice if Bluesky can automatically do this, or maybe it can remove this stupid `si` parameter since it serves no purpose other than provide YouTube with privacy-invading info. But whatever you do I'm happy with!

![highlighted.jpg](https://github.com/bluesky-social/social-app/assets/52135169/afbf15ec-addf-4ccf-9cdb-ad2605667f4a)

**Comment by mozzius** [comment]
yep, as Mary said, remove `si=xxx&` rather than `?si=xxx` and it will work (it's just how URLs work)
